lib/OptimizationManopt/src/OptimizationManopt.jl

```diff
@@ -285,19 +459,24 @@ function SciMLBase.__solve(cache::OptimizationCache{
 
     gradF = build_gradF(cache.f, cur)
 
+    hessF = build_hessF(cache.f, cur)
+
     if haskey(solver_kwarg, :stopping_criterion)
         stopping_criterion = Manopt.StopWhenAny(solver_kwarg.stopping_criterion...)
     else
         stopping_criterion = Manopt.StopAfterIteration(500)
     end
 
     opt_res = call_manopt_optimizer(manifold, cache.opt, _loss, gradF, cache.u0;
-        solver_kwarg..., stopping_criterion = stopping_criterion)
+        solver_kwarg..., stopping_criterion = stopping_criterion, hessF)
 
-    asc = get_active_stopping_criteria(opt_res.options.stop)
-
-    opt_ret = any(Manopt.indicates_convergence, asc) ? ReturnCode.Success :
+    if hasfield(typeof(opt_res.options), :stop)
```

Comment:
Why this check? All solvers _have_ such a stopping criterion and you can access it with `get_stopping_criterion`. https://manoptjl.org/stable/plans/state/#Manopt.get_stopping_criterion

The reason for the accessor is that the solver state (Manopt names, here maybe options?) might be wrapped (decorated) with record or debug, which themselves (as states) do not store a stop, but internally the have the original state and _that_ has a stop. So there is no reason to use a `Hatfield` here.